tests/test/Create2.t.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import "forge-std/Test.sol";
+
+contract C {
+    uint public num1;
+    uint public num2;
+
+    constructor(uint x, uint y) {
+        set(x, y);
+    }
+
+    function set(uint x, uint y) public {
+        num1 = x;
+        num2 = y;
+    }
+}
+
+contract Create2Test is Test {
```

Comment:
cool, I would also add the following test to verify the collision behavior:

```
    function test_create2_collision() public {
        C c1 = new C{salt: bytes32(uint256(42))}(1, 2);
        C c2 = new C{salt: bytes32(uint256(42))}(1, 2);
    }
```

Ideally it would do a proper EVM revert, currently we leak an internal exception: (`ValueError: existing address: 3149594625`)

Reply:
great suggestion for new test scenarios!  i added them.